File: 0539-minimum-time-difference/0539-minimum-time-difference.cpp
class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        int n = timePoints.size();
        vector<int> minutes;

        for(int i= 0; i<n; i++){
            string str = timePoints[i];

            int hour = stoi(str.substr(0,2));
            int minute = stoi(str.substr(3));
            int total = hour*60 + minute;
            minutes.push_back(total);
        }

        sort(minutes.begin(),minutes.end());

        int mini = INT_MAX;

        for(int i = 0; i<minutes.size()-1; i++){
            mini = min(mini,minutes[i+1]-minutes[i]);
        }

        mini = min(mini, minutes[0]+1440 - minutes[minutes.size()-1]);

        return mini;
    }
};

File: 0539-minimum-time-difference/0539-minimum-time-difference.java
class Solution {
    public int findMinDifference(List<String> timePoints) {
        ArrayList<Integer> minutes = new ArrayList<>();

        for(int i = 0; i<timePoints.size(); i++){
            int hour = Integer.parseInt(timePoints.get(i).substring(0,2));
            int minute = Integer.parseInt(timePoints.get(i).substring(3));
            int total = hour*60 + minute;
            minutes.add(total);
        }

        Collections.sort(minutes);
        int mini = Integer.MAX_VALUE;

        for(int i = 0; i<minutes.size()-1; i++){
            if(mini > (minutes.get(i+1) - minutes.get(i))){
                mini = (minutes.get(i+1) - minutes.get(i));
            }
        }

        if(mini > (minutes.get(0)+1440 - minutes.get(minutes.size()-1))){
            mini = (minutes.get(0)+1440 - minutes.get(minutes.size()-1));
        }
        
        return mini;
    }
}